torch/csrc/jit/symbolic_script.cpp

```diff
@@ -7,6 +7,331 @@ std::mutex lock;
+            self_size = self.size()
+            def backward(grad_output):
+                grad_self = AD_index_select_backward(grad_output, dim, result1, self_size, True)
+                return grad_self, None, None, None, None
+
+            return result0, result1, backward
+
+        def kthvalue(self,
+                     k: int,
+                     dim: int,
+                     keepdim: bool):
+            result0, result1 = torch.kthvalue(self, k, dim, keepdim)
+            self_size = self.size()
+            def backward(grad_output):
+                grad_self = AD_index_select_backward(grad_output, dim, result1, self_size, keepdim)
+                return grad_self, None, None, None
+
+            return result0, result1, backward
+
+        def AD_mm_backward_self(grad, mat2):
+            return grad.mm(mat2.t())
+
+        def AD_mm_backward_mat2(grad, self):
+            return self.t().mm(grad)
+
+        def mm(self, mat2):
+            def backward(grad_output):
+                grad_self = AD_mm_backward_self(grad_output, mat2)
+                grad_mat2 = AD_mm_backward_mat2(grad_output, self)
+                return grad_self, grad_mat2
+
+            return torch.mm(self, mat2), backward
+
+        def AD_permute_backward(grad,
+                                fwd_dims: List[int]):
+            ndims = len(fwd_dims)
+            dims = [0] * ndims
+
+            for i in range(ndims):
+                dims[fwd_dims[i]] = i
+
+            return grad.permute(dims)
+
+        def permute(self,
+                    dims: List[int]):
+            def backward(grad_output):
+                grad_self = AD_permute_backward(grad_output, dims)
+                return grad_self, None
+
+            return torch.permute(self, dims), backward
+
+        def AD_select_backward(grad,
+                               input_sizes: List[int],
+                               dim: int,
+                               index: int):
+            # FIXME: torchscript: torch.zeros(sizes, grad.options())
+            grad_input = torch.zeros(input_sizes).to(grad)
+            grad_input.select(dim, index).copy_(grad)
+            return grad_input
+
+        def select(self,
+                   dim: int,
+                   index: int):
+            self_size = self.size()
+            def backward(grad_output):
+                grad_self = AD_select_backward(grad_output, self_size, dim, index)
+                return grad_self, None, None
+
+            return torch.select(self, dim, index), backward
+
+        def AD_slice_backward(grad,
+                              input_sizes: List[int],
+                              dim: int,
+                              start: int,
+                              end: int,
+                              step: int):
+            # FIXME: torchscript: torch.zeros(sizes, grad.options())
+            grad_input = torch.zeros(input_sizes).to(grad)
```

Comment:
This looks like a pretty significant regressions. We shouldn't be implementing AD for functions that we can't handle efficiently

Reply:
@apaszke I can send a patch to temporarily comment out `select` and then try to figure out a fix on torchscript side.

Reply:
(slice is already commented)